packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -230,74 +213,127 @@ class _TreemapState extends State<Treemap> {
+            onRootChangedCallback: widget.onRootChangedCallback,
+          ),
+        ),
+      );
+    }
 
     // Construct pivot cell.
-    final pivotXCoord = isHorizontalRectangle ? x + list1Width : x + list2Width;
-    final pivotYCorrd = isHorizontalRectangle ? y : y + list1Height;
-    final pivotCell = TreemapCell(
-      key: UniqueKey(),
-      width: pivotBestWidth,
-      height: pivotBestHeight,
-      x: pivotXCoord,
-      y: pivotYCorrd,
-      node: pivotDataReference,
-      onTap: () => cellOnTap(rootNode.childrenMap[pivotDataReference.name]),
-      levelsVisible: widget.levelsVisible - 1,
+    final pivotXCoord = isHorizontalRectangle ? list1Width : list2Width;
+    final pivotYCoord = isHorizontalRectangle ? 0.0 : list1Height;
+
+    positionedTreemaps.add(
+      Positioned(
+        left: pivotXCoord,
+        top: pivotYCoord,
+        width: pivotBestWidth,
+        height: pivotBestHeight,
+        child: NewTreemap.fromRoot(
+          rootNode: pivotDataReference,
+          levelsVisible: widget.levelsVisible - 1,
+          onRootChangedCallback: widget.onRootChangedCallback,
+        ),
+      ),
     );
 
     // Construct list 3 sub-treemap.
-    final list3Ratio = list3ByteSize / totalSize;
+    final list3Ratio = list3ByteSize / totalByteSize;
     final list3Width = isHorizontalRectangle ? list3Ratio * width : width;
     final list3Height = isHorizontalRectangle ? height : list3Ratio * height;
     final list3XCoord =
-        isHorizontalRectangle ? x + list1Width + pivotBestWidth : x;
+        isHorizontalRectangle ? list1Width + pivotBestWidth : 0.0;
     final list3YCoord =
-        isHorizontalRectangle ? y : y + list1Height + pivotBestHeight;
-    final list3Cells = buildTreemap(
-      children: list3,
-      width: list3Width,
-      height: list3Height,
-      x: list3XCoord,
-      y: list3YCoord,
+        isHorizontalRectangle ? 0.0 : list1Height + pivotBestHeight;
+
+    if (list3.isNotEmpty) {
+      positionedTreemaps.add(
+        Positioned(
+          left: list3XCoord,
+          top: list3YCoord,
+          width: list3Width,
+          height: list3Height,
+          child: NewTreemap.fromNodes(
+            nodes: list3,
+            levelsVisible: widget.levelsVisible,
+            onRootChangedCallback: widget.onRootChangedCallback,
+          ),
+        ),
+      );
+    }
+
+    return positionedTreemaps;
+  }
+
+  Text buildNameAndSizeText({
+    @required Color fontColor,
+    @required bool onTwoLines,
+  }) {
+    final newline = onTwoLines ? '\n' : ' ';
+    return Text(
+      '${widget.rootNode.name}$newline${nodeSizeText()}',
+      style: TextStyle(color: fontColor),
+      textAlign: TextAlign.center,
+      overflow: TextOverflow.ellipsis,
     );
+  }
 
-    return list1Cells + [pivotCell] + list2Cells + list3Cells;
+  String nodeSizeText() {
+    final size = widget.rootNode.byteSize;
+    final sizeInKB = size / 1024;
+    if (sizeInKB < 1024.0) {
```

Comment:
make this a utility method to pretty display memory sizes. There are printMb and printKb methods already in src/utils.dart that have some overlap. I agree that for this case you do want to sometimes display KB and sometimes display MB. The existing helpers in src/utils/dart handle a couple more edge cases to make the output easier to read so you should use that logic when it comes to displaying values in KB and MB.

Reply:
To clarify, should I be adding this function to utils.dart? I do use the printKb and printMb functions to actually print the values, but I manually compute the size in KB first for the purpose of checking the size to decide whether to display the values in KB or MB.

Reply:
perhaps create a utility method `prettyPrintBytes(int size)` that leverages the existing `printMb` and `printKb` methods